glean-core/rlb/src/private/string.rs

```diff
@@ -28,6 +28,11 @@ impl StringMetric {
     pub fn new(meta: glean_core::CommonMetricData) -> Self {
         Self(Arc::new(glean_core::metrics::StringMetric::new(meta)))
     }
+
+    /// The internal constructor used by `LabeledMetric`.
+    pub(crate) fn new_from_meta(metric: Arc<glean_core::metrics::StringMetric>) -> Self {
```

Comment:
Do we really need this method?
We could just as well do `StringMetric(metric)` where it is called.

Or we could `impl From<glean_core::metrics::StringMetric> for StringMetric` that then puts it in the `Arc` and returns `StringMetric`.

At the very least the name here is wrong. It's not constructing something new from `meta` (throughout the codebase we use the name `meta` only for `CommonMetricData` so far)